src/Apps/Collect/Components/Collection/Header/__tests__/DefaultHeader.test.tsx

```diff
@@ -0,0 +1,95 @@
+import { defaultCollectionHeaderArtworks } from "Apps/Collect/Components/Collection/Header/__tests__/fixtures/artworks"
+import {
+  CollectionDefaultHeader,
+  getHeaderArtworks,
+} from "Apps/Collect/Components/Collection/Header/DefaultHeader"
+import { shallow } from "enzyme"
+import React from "react"
+import renderer from "react-test-renderer"
+
+describe("artworks", () => {
+  it("returns an array of repeating artworks when original array of works is too small to fill header ", () => {
+    const artworks = defaultCollectionHeaderArtworks.hits.slice(0, 3)
+    const result = getHeaderArtworks(artworks, 1308, false)
+
+    expect(result.length).toBeGreaterThan(artworks.length)
+    expect(result).toHaveLength(6)
+  })
+
+  it("returns an array of artworks of equal length as the original array received when header can fit all works ", () => {
+    const artworks = defaultCollectionHeaderArtworks.hits
+
+    const result = getHeaderArtworks(artworks, 1308, false)
+    expect(artworks.length).toEqual(result.length)
+    expect(result).toHaveLength(10)
+  })
+})
```

Comment:
There's one more test case we could add that would be really valuable. `it("returns only the number of artworks necessary to fill the header at a small resolution")`. 

If a user is on a mobile device, they are going to only need a few images from the collection. Giving them all of the originals will result in them downloading a handful of images that they'll never be able to see.

I don't think the current code would pass this test, but I think adding this test & refactoring to support it might simplify the implementation. We would iterate through all images until it was wide enough to exceed the viewport width, and start over in the list of images when we reach the end.

Reply:
Great point @pepopowitz +1 to that

Reply:
Yep, decided to just loop through once, only adding enough artworks to fill the header regardless of the screen size.

Reply:
Do you mind adding a test case to capture this edge-case?